shared/middleware/helpers/source_bucket.rb

```diff
@@ -66,6 +66,63 @@ def restore_previous_version(encrypted_channel_id, filename, version_id, user_id
     response.to_h
   end
 
+  # Copies the files in the src_channel to the dest_channel
+  # Update the animation manifest to include the version ids of the animations
+  # in dest_channel
+  # Note: this function assumes that the animations have already been copied
+  def remix_source(src_channel, dest_channel, options={})
+    src_owner_id, src_channel_id = storage_decrypt_channel_id(src_channel)
+    dest_owner_id, dest_channel_id = storage_decrypt_channel_id(dest_channel)
+
+    src_prefix = s3_path src_owner_id, src_channel_id
+
+    # For each file, copy it, update the animations, return it
+    result = s3.list_objects(bucket: @bucket, prefix: src_prefix).contents.map do |fileinfo|
+      filename = %r{#{src_prefix}(.+)$}.match(fileinfo.key)[1]
+      next unless (!options[:filenames] && (!options[:exclude_filenames] || !options[:exclude_filenames].include?(filename))) || options[:filenames].try(:include?, filename)
+      mime_type = Sinatra::Base.mime_type(File.extname(filename))
+      category = mime_type.split('/').first  # e.g. 'image' or 'audio'
```

Comment:
`mime_type` and `category` only get used in the return value, so they can go too.